### ISSUE TITLE
Question: How to graceful stop docker-parity?

### ISSUE THREAD

**Issue by isLishude** [paritytech/parity-ethereum#10053]
_Before filing a new issue, please **provide the following information**._

- **Parity Ethereum version**: stable
- **Operating system**: Linux
- **Installation**: homebrew / one-line installer / built from source
docker-compose file

```yml
version: "3"

services:

  parity:
    image: "parity/parity:stable"
    restart: unless-stopped
    ports:
      - "30303:30303"
      - "30303:30303/udp"
      - "8545:8545"
    volumes:
      - ${PWD}/.parity:/home/parity/.local
    stop_signal: SIGINT
    stop_grace_period: 2m
    command:
      - --no-ui
      - --no-dapps
      - --no-config
      - --no-ws
      - --tracing=on
      - --auto-update=none
      - --max-peers=100
      - --jsonrpc-interface=all
      - --jsonrpc-apis=safe
      - --jsonrpc-hosts=all
      - --jsonrpc-threads=8
      - --jsonrpc-cors=all
      - --jsonrpc-server-threads=8
      - --jsonrpc-max-payload=10
      # - --tx-queue-size=8192
      # - --tx-queue-per-sender=16
      # - --tx-queue-mem-limit=4

    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"
```

I setted `stop_signal: SIGINT` && `stop_grace_period: 2m` but it does not work.

- **Fully synchronized**: no
- **Network**: ethereum
- **Restarted**: yes

_Your issue description goes here below. Try to include **actual** vs. **expected behavior** and **steps to reproduce** the issue._

**Comment by isLishude** [comment]
![image](https://user-images.githubusercontent.com/24730006/49871638-bc508600-fe51-11e8-9bd9-36d4b9a2a626.png)
![image](https://user-images.githubusercontent.com/24730006/49871652-c4a8c100-fe51-11e8-9be4-729532706e11.png)

When I run `docker-compose down` ,it still syncs blocks. And after 2m stop by SIGKILL.

![image](https://user-images.githubusercontent.com/24730006/49871772-10f40100-fe52-11e8-8232-a4617efa1d95.png)

by the way,My version is latest stable v2.1.9

**Comment by joshua-mir** [comment]
Do you see `finishing work, please wait` anywhere in your logs? This might be related to a known issue; there's a fix for it in beta https://github.com/paritytech/parity-ethereum/pull/9886

**Comment by isLishude** [comment]
No any logs about `finishing work`.

I upgraded parity to v2.1.4 but does not work either

**Comment by joshua-mir** [comment]
Beta is 2.2.x currently, not 2.1.4 😅

**Comment by isLishude** [comment]
a typo v2.2.4

**Comment by isLishude** [comment]
![image](https://user-images.githubusercontent.com/24730006/49873001-bceb1b80-fe55-11e8-800b-ebd9727266aa.png)

**Comment by joshua-mir** [comment]
Parity doesn't appear to be receiving SIGINT for some reason... are you seeing the same behavior when manually running `docker stop`? 

Pinging @ddorgan because I'm not familiar enough with docker compose to tell you if anything is obviously wrong here.

**Comment by isLishude** [comment]
I am not good at shell script,maybe is right.

**Comment by isLishude** [comment]
@joshua-mir I make a PR  to fix this issue,could you review this PR #10059?
